tests/data-fetch-mixin.js

```diff
@@ -197,8 +197,48 @@ describe("DataFetch mixin", function() {
     var prevCallCount = fakeComponent.setState.callCount;
 
     var onError = $.ajax.args[0][0].error;
-    onError(null, 503, 'foobar');
+    onError({}, 503, 'foobar');
 
     expect(fakeComponent.setState.callCount).to.equal(prevCallCount);
-  })
+  });
+
+  it("should set isFetchingData to false in initial state", function() {
+    var initialState = fakeComponent.getInitialState();
+
+    expect(initialState.isFetchingData).to.equal(false);
+  });
+
+  it("should set dataError to null in initial state", function() {
+    var initialState = fakeComponent.getInitialState();
+
+    expect(initialState.dataError).to.equal(null);
+  });
+
+  it("should set dataError for a failed request", function() {
```

Comment:
You're actually testing `statusText` only in this test. You should have separate tests for the other 3 keys of `dataError` as well.

This means you will also have to mock the xhr object with a `status` key and the `err` to be an object with a `toString` key with a spy attached.

You can wrap all `dataError` related tests in a nested `describe` block if a common `beforeEach` would help.